logic/hosts.go

```diff
@@ -313,3 +318,60 @@ func GetRelatedHosts(hostID string) []models.Host {
 	}
 	return relatedHosts
 }
+
+// CheckHostPort checks host endpoints to ensures that hosts on the same server
+// with the same endpoint have different listen ports
+// in the case of 64535 hosts or more with same endpoint, ports will not be changed
+func CheckHostPorts(h *models.Host) {
+	h.ListenPort = checkPort(h, h.ListenPort)
+	h.ProxyListenPort = checkPort(h, h.ProxyListenPort)
+	// rerun if ports were set to same value
+	if h.ListenPort == h.ProxyListenPort {
+		updatePort(&h.ProxyListenPort)
+		h.ProxyListenPort = checkPort(h, h.ProxyListenPort)
+	}
+}
+
+func checkPort(h *models.Host, p int) int {
+	currentPort := h.ListenPort
```

Comment:
if you set the proxylistenport to p, it will default to listenport? is this purposeful?

Reply:
good catch.... that's what happens when you try to reduce code copying by combining two similar functions into one. 
fixed.